assets/src/application/ui/layout/espressoDropdownMenu/Menu.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { Children, cloneElement } from 'react';
+import classNames from 'classnames';
+import { DOWN } from '@wordpress/keycodes';
+import { Dropdown, NavigableMenu } from '@wordpress/components';
+
+import { EspressoButton } from '@application/ui/input';
+
+import { EspressoDropdownMenuProps as Props } from './types';
+
+import './style.scss';
+
+const mergeProps = (defaultProps: Props, props: Props) => {
+	const mergedProps = {
+		...defaultProps,
+		...props,
+	};
+
+	if (props?.className && defaultProps?.className) {
+		mergedProps.className = classNames(props?.className, defaultProps?.className);
+	}
+
+	return mergedProps;
+};
+
+const EspressoDropdownMenu: React.FC<Props> = ({
+	children,
+	className,
+	icon = 'menu',
+	label,
+	popoverProps = { position: 'top center' },
+	toggleProps,
+	menuProps,
+}) => {
+	const mergedPopoverProps = mergeProps(
+		// @ts-ignore
+		{
+			className: 'components-dropdown-menu__popover',
+		},
+		popoverProps
+	);
+
+	return (
+		<Dropdown
+			className={classNames(' components-dropdown-menu', className)}
+			popoverProps={mergedPopoverProps}
+			renderToggle={({ isOpen, onToggle }) => {
```

Comment:
the callback passed here could likely be extracted to its own component which would better adhere to SRP (same with renderContent)

Reply:
done

Reply:
for simplicity and since, those 2 components aren't exposed, I've named those:
`Content` and `Toggle`, based on the prop that is accepting those